### DISCUSSION TITLE
Is there an API koreader provides that I can use to query recently read books?

### DISCUSSION THREAD

**Discussion by mainrs** [koreader/koreader#11704] [Q&A]
I want to aggregate books that I read from various sources/apps for a custom launcher I am working on.

Usually, what I have seen, is that apps either provide an Intent-based API or a ContentProvider. 

This requires access to the database/data storage mechanism koreader uses inside of Android Kotlin classes. Can this be done? I.e., asking the Lua code for information inside the database from Kotlin?

**Comment by hius07** [comment]
You can take koreader/history.lua and parse it yourself.

**Reply by hius07** [reply]
But... It does not contain books' metadata, only filepath and datetime.
